reconcile.go

```diff
@@ -32,18 +32,30 @@ func (a *Autopilot) reconcile() error {
 	// have the promoter calculate the required Raft changeset.
 	changes := a.promoter.CalculatePromotionsAndDemotions(conf, state)
 
+	// Apply the demotions to the failed servers first, but only once per a
+	// reconciliation round and if the number of voters is odd. In that case we
+	// don't want to start with the promotions, since that temporalily inflates
+	// quorum, which could lead to cluster failure. If we did apply any, then stop
+	// here as we do not want to apply the promotions at the same time as a means of
+	// preventing cluster instability.
+	if len(state.Voters)%2 != 0 {
+		if done, err := a.applyDemotions(state, changes, true); done {
+			return err
+		}
```

Comment:
I agree. The reason I made the logic stop the reconcile round after a single demotion, was because I was trying to adhere to the current practice in this part of code. Having thought about it some more now, I also do not see a reason to wait for the next reconcile loop to promote healthy nodes. 

I will update the code for now and we can always change it again if we find a valid reason to keep demotion of failed node and promotions in the same round.

Reply:
Fixed in 165e61e5d149b52eb40db3756a8af0fd4f414d75 (I am cleaning up the threads on the PR, hence this comment now 😄)